Link libdl in core/platform/default/stacktrace

Flamefire opened the pull request:
~Required due to use of `dladdr` via the `stacktrace_handler` dependency~

Required due to use of `dladdr` in `stacktrace.h` which is e.g. a dependency of `test_main` via `stacktrace_handler`

Fixes #45013

Flamefire committed:
Link libdl in core/platform/test_main

Required due to use of `dladdr` via the `stacktrace_handler` dependency
Fixes #45013

Flamefire committed:
Fix formatting

Flamefire commented:
Seems CI wants a slightly different formatting. Added that change (only), the other failures are unrelated

cantonios commented on tensorflow/core/platform/BUILD:
```diff
@@ -1095,7 +1095,10 @@ cc_library(
-        "//conditions:default": ["-lm"],
+        "//conditions:default": [
+            "-lm",
+            "-ldl",
```
The flag should probably be added to linkopts here instead: https://github.com/tensorflow/tensorflow/blob/71738a2bfb3824611debc6c54dc26c50072473a0/tensorflow/core/platform/default/BUILD#L376

Flamefire replied:
Makes sense to me too. Added a commit doing that. Quick test shows that this works.

gbaned commented:
@Flamefire Can you please check @cantonios's comments and keep us posted ? Thank you!

Flamefire commented:
Testing that now, thanks!

Flamefire committed:
Move linking ldl to the stacktrace library

`stacktrace.h` uses `dladdr` so that is the correct place to add the
dependency on libdl.

Flamefire commented:
@gbaned Tested and rebased to implement the suggested change and resolve a conflict that occurred in the meantime.

Flamefire commented:
Unrelated CI failures, e.g.:

> ERROR: The project you're trying to build requires Bazel 5.3.0 (specified in /workspace/.bazelversion), but it wasn't found in /usr/local/lib/bazel/bin.
